OMPython/__init__.py

```diff
@@ -447,3 +462,727 @@ def getClassNames(self, className=None, recursive=False, qualified=False, sort=F
+					index_ = self.pNamesList.index(paraName)
+					return self.pValuesList[index_]
+				else:
+					print '!!! ', paraName, ' does not exist'
+					return None
+			except Exception as e:
+				print e
+				return None
+
+	def getInputValues(self, iName=None):
+		if iName is None:
+			return self.inputsVal
+		else:
+			if iName in self.iNamesList:
+				index_ = self.iNamesList.index(iName)
+				return self.inputsVal[index_]
+			else:
+				print '!!! ', iName, ' does not exist'
+				return None
+
+	def getOutputValues(self, oName=None):
+		if oName is None:			
+			if self.oValuesList == []:
+				for l in self.quantitiesList:
+					if(l.causality == "output"):
+						self.oValuesList.append(l.start)
+			return self.oValuesList
+		else:
+			if oName in self.oNamesList:
+				index_ = self.oNamesList.index(oName)
+				return self.oValuesList[index_]
+			else:
+				print '!!! ', oName, ' does not exist'
+				return None
+
+	def getSimulationValue(self):
+		if self.simValuesList == []:
+			root = self.tree.getroot()
+			rootGSV = self.root
+			for attr in rootGSV.iter('DefaultExperiment'):
+				startTime = attr.get('startTime')
+				self.simValuesList.append(startTime)
+				stopTime = attr.get('stopTime')
+				self.simValuesList.append(stopTime)
+				stepSize = attr.get('stepSize')
+				self.simValuesList.append(stepSize)
+				tolerance = attr.get('tolerance')
+				self.simValuesList.append(tolerance)
+				solver = attr.get('solver')
+				self.simValuesList.append(solver)
+		return self.simValuesList
+	
+	def getSimulationOptions(self):
+		dic = {'Options' : self.simNamesList, 'Value' : self.simValuesList}
+		frame = pd.DataFrame(dic)		
+		return frame
+	
+	def getOptimizationOptions(self):
+		dic = {'Options' : self.optimizeOptionsNamesList, 'Value' : self.optimizeOptionsValuesList}
+		frame = pd.DataFrame(dic)
+		return frame
+	
+	def getLinearizationOptions(self):
+		dic = {'Options' : self.linearizeOptionsNamesList, 'Value' : self.linearizeOptionsValuesList}
+		frame = pd.DataFrame(dic)
+		return frame
+
+	def simulate(self):
+		if (self.inputFlag == True):
+			self.simInput()
+			getExeFile = '{}.{}'.format(self.modelName, "exe")
+			check_exeFile_ = os.path.exists(getExeFile)
+			if(check_exeFile_):
+				cmd = getExeFile + " -csvInput=" + self.csvFile
+				
+				subprocess.call(cmd, shell = False)
+				return
+			else:
+				print "Error: application file not generated yet"
+				return None
+	
+		else:
+			getExeFile = '{}.{}'.format(self.modelName, "exe")
+			check_exeFile_ = os.path.exists(getExeFile)
+			if(check_exeFile_):
+				cmd = getExeFile
+				subprocess.call(cmd, shell = False)
+				return
+			else:
+				print "Error: application file not generated yet"
+				return None
+				
+	def getSolutions(self, varList):
+		for v in varList:
+			if v == 'time':
+				continue
+			isExist = False
+			for l in self.quantitiesList:
+				if v == l.name:
+					isExist = True
+				else:
+					continue
+			if(isExist == False):
```

Comment:
afaict, the `for` and `isExist` can be dropped, and the `if` be replaced with
`if v not in [l.name for l in self.quantitiesList]:`